pdns/syncres.cc

```diff
@@ -714,22 +714,30 @@ int SyncRes::doResolve(const DNSName &qname, const QType &qtype, vector<DNSRecor
     if (qtype == QType::DS) {
       nsdomain.chopOff();
     }
-    // the two retries allow getBestNSFromCache&co to reprime the root
-    // hints, in case they ever go missing
-    for (int tries = 0; tries < 2 && bestns.empty(); ++tries) {
-      bool flawedNSSet = false;
-      set<GetBestNSAnswer> beenthereIgnored;
-      getBestNSFromCache(nsdomain, qtype, bestns, &flawedNSSet, depth, beenthereIgnored);
-    }
 
-    if (bestns.size() == 0) {
-      // Something terrible is wrong
-      QLOG("Step1 No ancestor found return ServFail");
-      return RCode::ServFail;
+    DNSName fwdomain(qname);
+    bool forwarded = getBestAuthZone(&fwdomain) != t_sstorage.domainmap->end();
+    if (forwarded) {
+      QLOG("Step1 qname is in a forwarded domain");
```

Comment:
Couldn't that domain be one from a local auth zone as well?

Reply:
I do not think so, since for auth and recursive forwards QM is not used at all.

Reply:
Right!